ui/src/main/java/org/pentaho/di/ui/core/events/dialog/SelectionAdapterOptions.java

```diff
@@ -21,13 +21,16 @@
  ******************************************************************************/
 package org.pentaho.di.ui.core.events.dialog;
 
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
 
 public class SelectionAdapterOptions {
   private SelectionOperation selectionOperation;
   private String[] filters;
   private String defaultFilter;
   private String[] providerFilters;
+  private List<ConnectionFilterType> connectionFilters = new ArrayList<>();
   private boolean useSchemaPath;
 
   public SelectionAdapterOptions( SelectionOperation selectionOperation, String[] filters, String defaultFilter,
```

Comment:
I didn't like the ever growing constructor.  I'd actually like to propose the builder pattern or just basic POJO instead of these really long constructors.